### PR TITLE
Add feature to star / unstar contacts (favourites list)

### LINKED ISSUE #79: Add contacts to favourites list
- Add StarredStatus attribute
- Add command star & unstar [index] OR [full name] to mark or unmark contact as favourite
- Modify list command so list * will list starred contacts

Potentially:
- Represent favourite contacts using GUI (e.g. star)

### PR DESCRIPTION
**Pull request by myrilla12**
closes #79 
Ready for review

### Description
Users can now star and unstar contacts using commands `star [index/name]` and `unstar [index/name]`. Users can view their list of starred contacts using command `list *`.

### Key Changes
- Add attribute StarredStatus to Person model
- Add StarCommand and UnstarCommand and their respective parsers
- Update ListCommand to allow listing of starred contacts
- Update & add tests
- Update features in user guide

### Potential Enhancement
- Use GUI to represent starred contact on PersonCard

### CODE REVIEW

**Comment by LynetteLeeSiYing** [comment]
LGTM!!

File: src/main/java/seedu/address/logic/commands/ListCommand.java
Comment (ritvi12) [inline src/main/java/seedu/address/logic/commands/ListCommand.java:1]: LGTM!

**Review by Bonzzz3 — APPROVED** [review]
LGTM!

**Review by ritvi12 — COMMENTED** [review]
LGTM!

**Review by ritvi12 — APPROVED** [review]
The code logic seems good!

**Pull request merged**